drivers/hmis_external_apis/extensions/hmis/unit_type_extension.rb

```diff
@@ -32,9 +32,13 @@ def track_availability(project_id:, user_id:)
           user_id: user_id,
           unit_type_id: id,
         )
-        HmisExternalApis::AcHmis::UpdateUnitAvailabilityJob
-          .set(wait: 1.minute) # short wait to accumulate batch of changes before update
-          .perform_later
+
+        # Don't re-queue if job is already queued
+        return if Delayed::Job.queued?('HmisExternalApis::AcHmis::UpdateUnitAvailabilityJob')
```

Comment:
Looks like there's a chance for race conditions here. Maybe it doesn't matter if this is called frequently and so dropped jobs are just delayed for a short time.

Reply:
I can't remember, but I think `queued?` looks for one that hasn't started.  If it has started, it should queue up the next one, so I can see a situation where there may be more than one in the queue, I guess we should make sure they are using an advisory lock so they don't try to process the same data at the same time.  

trying to determine if there's a race condition where the job gets picked up while it's checking, so it appears queued, but is actually running and for whatever reason misses the newly dirtied records.....

Reply:
Yes, `queued?` only looks for jobs that have started.

If there is a potential race condition here, then it would also affect the service history processing that queues in an after_save hook using the same mechanism.


I'm going to go ahead and merge this, as I don't think it's a concern if there is an occasional delay with unit changes.